src/ORM/DB.php

```diff
@@ -60,7 +61,22 @@ class DB
      */
     protected static $configs = [];
 
+    /**
+     * Array of classes that have been confirmed ready for database queries.
+     * When the database has once been verified as ready, it will not do the
+     * checks again.
+     *
+     * @var boolean[]
+     */
+    private static $database_is_ready = [];
 
+    /**
+     * If set to a value for a given class, {@link database_is_ready()}
+     * will always return that value. Used for unit testing.
+     *
+     * @var boolean[]
+     */
+    private static $force_database_is_ready = [];
```

Comment:
I don't like the need for *both* the `$database_is_ready` and `$force_database_is_ready` arrays. Can't `force_database_is_ready()` just push into the `$database_is_ready` array thus preventing a "re-check" when `database_is_ready` is called? or does this have too significant an impact on the expectation of tests?

Reply:
Yup, fair point. I was mostly doing a lift-and-shift from Security but there's no harm (that I can think of) consolidating the two.